databricks-sdk-java/src/main/java/com/databricks/sdk/core/NotebookNativeCredentialsProvider.java

```diff
@@ -0,0 +1,133 @@
+package com.databricks.sdk.core;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A CredentialsProvider that uses the API token from the command context to authenticate.
+ *
+ * <p>The token and hostname are read from the command context, which can be retrieved through the
+ * dbutils API. As the Java SDK does not depend on DBUtils directly, reflection is used to retrieve
+ * the token. This token should be available wherever the DBUtils API is accessible (i.e. in the
+ * Spark driver).
+ */
+public class NotebookNativeCredentialsProvider implements CredentialsProvider {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(NotebookNativeCredentialsProvider.class);
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  @Override
+  public String authType() {
+    return "runtime";
+  }
+
+  @Override
+  public HeaderFactory configure(DatabricksConfig config) {
+    if (System.getenv("DATABRICKS_RUNTIME_VERSION") == null) {
```

Comment:
I think instead of checking this here, we should have an attribute in config which would be set at resolve() and then check that attribute here. We do that for other providers.

Reply:
Interesting. I was mainly copying the implementation from https://github.com/databricks/databricks-sdk-py/blob/main/databricks/sdk/core.py#L97-L98, which just reads this environment variable from the actual environment. I suppose we could add this to the configuration itself in all SDKs, need to think about it a bit more.

Reply:
Thinking about this a bit more, I think we should keep this as-is. The main issue is that this environment variable is not expected to be configured or set by users, it's just a way that we detect whether we're running in DBR.